### PR TITLE
Add more filespace/tablespace tests to installcheck.

### PR DESCRIPTION
**Pull request by jimmyyih**
This commit adds more filespace and tablespace test coverage in installcheck.
Most of these test additions are modified and inspired from Pivotal's own
internal tests.

### CODE REVIEW

**Comment by atris** [comment]
+1

On Thu, Mar 24, 2016 at 1:48 AM, Jimmy Yih <notifications@github.com> wrote:

> This commit adds more filespace and tablespace test coverage in
> installcheck.
> Most of these test additions are modified and inspired from Pivotal's own
> internal tests.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/greenplum-db/gpdb/pull/565
> Commit Summary
>
>    - Add more filespace/tablespace tests to installcheck.
>
> File Changes
>
>    - *M* src/test/regress/input/filespace.source
>    <https://github.com/greenplum-db/gpdb/pull/565/files#diff-0> (1282)
>    - *M* src/test/regress/output/filespace.source
>    <https://github.com/greenplum-db/gpdb/pull/565/files#diff-1> (1584)
>
> Patch Links:
>
>    - https://github.com/greenplum-db/gpdb/pull/565.patch
>    - https://github.com/greenplum-db/gpdb/pull/565.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/565>
>



-- 
Regards,

Atri
*l'apprenant*

File: src/test/regress/input/filespace.source [hunk 1]
```diff
@@ -123,3 +123,1285 @@ drop filespace regressionfs1;
...
+ALTER DATABASE fsts_db_name1 RENAME TO fsts_new_db_name1;
+ALTER DATABASE fsts_new_db_name1 RENAME TO fsts_db_name1;
+\c fsts_db_name1
+SET default_tablespace = 'regression_ts_b3';
+
+-- Create heap list partition table
+CREATE TABLE fsts_test_part5 (id int, name text, rank int, year date, gender char(1)) TABLESPACE regression_ts_a3 DISTRIBUTED BY (id)
+  PARTITION BY LIST (gender) SUBPARTITION BY RANGE (year) SUBPARTITION TEMPLATE (start (date '2001-01-01')) (values ('M'), values ('F'));
+
+-- Create an index for the creation of block dir
```
Comment (ashwinstar) [inline src/test/regress/input/filespace.source]: block dir is only used for AO tables so comment seems incorrect, unless it has some other meaning here.

File: src/test/regress/input/filespace.source [hunk 2]
```diff
@@ -123,3 +123,1285 @@ drop filespace regressionfs1;
...
+ALTER TABLE fsts_test_part5_1_prt_1_2_prt_1 SET TABLESPACE regression_ts_c2;
+
+-- Insert and then truncate table
+INSERT INTO fsts_test_part5 VALUES (5,'toki',5,'2003-01-01','M');
+TRUNCATE fsts_test_part5;
+
+-- Select from the table
+SELECT * FROM fsts_test_part5;
+
+-- Alter Tbale add column
```
Comment (ashwinstar) [inline src/test/regress/input/filespace.source]: s/Tbale/Table

**Comment by ashwinstar** [comment]
Would have been great if we had comment where one scenario starts and what it does, instead of  comment for every sql statement which really is not adding too much value as its says REINDEX followed by REINDEX command. We need to get better at having comments more for why vs what. But know this is legacy code so its okay to get it in in current form if big effort to add the same :+1:

**Comment by jimmyyih** [comment]
Changes made relative to Ashwin's comments.  Pushed into master: https://github.com/greenplum-db/gpdb/commit/23ab1bd5617fbc6bb9f156a53e221eb39eb65f92

**Pull request closed without merging**